cls_module/cls_module/components/local_optimizer.py

```diff
@@ -0,0 +1,29 @@
+from torch.optim.optimizer import Optimizer, required
+
+class LocalOptim(Optimizer):
+  def __init__(self, named_params, lr=required):
+    self.param_names, params = zip(*named_params)
+
+    if lr is not required and lr < 0.0:
```

Comment:
Looks like this should be:
`if lr is required and lr < 0.0`
i.e. if if it is not required, then it wouldn't matter if it was below 0

Reply:
or rather, since we know that we DO need lr, should it instead of an OR instead of an AND between these two

Reply:
The conditional is checking that variable `lr` is NOT defaulting to the default value of `required`, which is how PyTorch deals with 'required' fields.

This is essentially saying: If an `lr` was passed, and its value is `< 0` then, trigger the error.

If nothing was passed, PyTorch automatically triggers the validation of the required field.